pkg/devfile/generator/utils.go

```diff
@@ -46,7 +46,12 @@ func convertPorts(endpoints []v1.Endpoint) []corev1.ContainerPort {
 		} else {
 			portProtocol = corev1.ProtocolTCP
 		}
-		name := fmt.Sprintf("%d-%s", portNumber, strings.ToLower(string(portProtocol)))
+		name := endpoint.Name
```

Comment:
We are changing a user provided name underneath the covers if length exceeds 15 characters.  It'll be good to warn them of these changes ahead of time (via cmd line, log, and/or doc)

Reply:
yes. This change is because we want to unify the service port name & the container port name. since service port does not accept starting with a number (has to be alphabet), thus we use the convention `port-<number>`
@valaparthvi  FYI.

Reply:
Does the warning need to happen on the `odo` side?

Reply:
correct